pkg/clusters/cleanup.go

```diff
@@ -114,6 +117,88 @@ func (c *Cleaner) Cleanup(ctx context.Context) error {
 	return g.Wait()
 }
 
+// fixupObjKinds takes a client.Object and checks if it's of one of the gateway
+// API types and if so then it adjusts that object's Kind and APIVersion.
+// The reason we need this is that when interacting via Gateway API objects
+// for some reason .TypeMeta.Kind and .TypeMeta.APIVersion are empty.
```

Comment:
Are we sure that's behavior specific to Gateway API objects only? 🤔 What would be the reason for that?

Reply:
I was afraid I'm gonna fall into a rabbit whole of sorts so I just left it at that :)

Reply:
Alright. I was just curious if that could be done in a more generic way with a clever dose of reflection. Let's not fall into that hole for now. 👍

Reply:
I've added links to issues in kubernetes that I believe are related.